plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -300,6 +329,52 @@ public IcebergTableHandle getTableHandle(ConnectorSession session, SchemaTableNa
                 ImmutableList.of());
     }
 
+    @Override
+    public boolean isSupportedVersionType(ConnectorSession session, SchemaTableName tableName, PointerType pointerType, io.trino.spi.type.Type versioning)
+    {
+        switch (pointerType) {
+            case TEMPORAL:
+                return versioning instanceof TimestampWithTimeZoneType || versioning instanceof TimestampType;
+            case TARGET_ID:
+                return versioning == BIGINT;
+        }
+        return false;
+    }
+
+    private long getSnapshotIdFromVersion(Table table, ConnectorTableVersion version)
+    {
+        io.trino.spi.type.Type versionType = version.getVersionType();
+        switch (version.getPointerType()) {
+            case TEMPORAL:
+                verify(versionType instanceof TimestampWithTimeZoneType || versionType instanceof TimestampType,
+                        "Iceberg temporal time travel only supports types TimestampWithTimeZone and Timestamp, but got " + versionType);
```

Comment:
remove `|| versionType instanceof TimestampType` since it's not supported (yet)

also, `TrinoException` should be raised (not `VerifyException`), since it's a user error